webapp/src/main/java/org/apache/atlas/util/AccessAuditLogsIndexCreator.java

```diff
@@ -0,0 +1,355 @@
+        }
+        if (client != null) {
+            try {
+                Request request = new Request("HEAD", index);
+                Response response = client.performRequest(request);
+
+                int statusCode = response.getStatusLine().getStatusCode();
+                if (statusCode == 200) {
+                    LOG.info("Entity audits index exists!");
+                    exists = true;
+                } else {
+                    LOG.info("Entity audits index does not exist!");
+                    exists = false;
+                }
+
+            } catch (Exception e) {
+                LOG.info("Index " + this.index + " not available.");
+            }
+            if (!exists) {
+                LOG.info("Index does not exist. Attempting to create index:" + this.index);
+                try {
+                    HttpEntity entity = new NStringEntity(es_ranger_audit_schema_json, ContentType.APPLICATION_JSON);
+                    Request request = new Request("PUT", index);
+
+                    /*if (this.no_of_shards >= 0 && this.no_of_replicas >= 0) {
+                        request.settings(Settings.builder().put("number_of_shards", this.no_of_shards)
+                                .put("number_of_replicas", this.no_of_replicas));
+                    }*/
+
+                    request.setEntity(entity);
+                    Response response = client.performRequest(request);
+
+                    if (response != null && response.getStatusLine().getStatusCode() == 200) {
+                        LOG.info("Index " + this.index + " created successfully.");
+                        exists = true;
+                    }
+                } catch (Exception e) {
+                    LOG.error("Unable to create Index. Reason:" + e.toString());
+                    e.printStackTrace();
+                }
+            } else {
+                LOG.info("Index " + this.index + " is already created.");
+            }
+        }
+        return exists;
+    }
+
+    private void logErrorMessageAndWait(String msg, Exception exception) {
+        retry_counter++;
+        String attemptMessage;
+        if (max_retry != TRY_UNTIL_SUCCESS) {
+            attemptMessage = (retry_counter == max_retry) ? ("Maximum attempts reached for setting up elasticsearch.")
+                    : ("[retrying after " + time_interval + " ms]. No. of attempts left : "
+                    + (max_retry - retry_counter) + " . Maximum attempts : " + max_retry);
+        } else {
+            attemptMessage = "[retrying after " + time_interval + " ms]";
+        }
+        StringBuilder errorBuilder = new StringBuilder();
+        errorBuilder.append(msg);
+        if (exception != null) {
+            errorBuilder.append("Error : ".concat(exception.getMessage() + ". "));
+        }
+        errorBuilder.append(attemptMessage);
+        LOG.error(errorBuilder.toString());
+        try {
+            Thread.sleep(time_interval);
```

Comment:
Why holding the main thread here?

Reply:
I have marked this class as Thread so this all runs in separate thread & will not block main thread